src/runtime/hexagon/tests/run_all_tests.cc

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#if defined(__hexagon__)
+
+#include <gtest/gtest.h>
+#include <tvm/runtime/packed_func.h>
+#include <tvm/runtime/registry.h>
+
+#include <string>
+#include <vector>
+
+#include "../src/support/utils.h"
+
+namespace tvm {
+namespace runtime {
+namespace hexagon {
+
+TVM_REGISTER_GLOBAL("hexagon.run_all_tests").set_body([](TVMArgs args, TVMRetValue* rv) {
+  // gtest args are passed into this packed func as a singular string
+  // split gtest args using <space> delimiter and build argument vector
+  std::vector<std::string> parsed_args = tvm::support::Split(args[0], ' ');
+  std::vector<char*> argv;
+
+  // add executable name
+  argv.push_back(const_cast<char*>("hexagon_run_all_tests"));
+
+  // add parsed arguments
+  for (int i = 0; i < parsed_args.size(); ++i) {
+    argv.push_back(const_cast<char*>(parsed_args[i].data()));
+  }
+
+  // end of parsed arguments
+  argv.push_back(nullptr);
+
+  // set argument count
+  int argc = argv.size() - 1;
+
+  // initialize gtest with arguments and run
+  ::testing::InitGoogleTest(&argc, argv.data());
+  *rv = RUN_ALL_TESTS();
```

Comment:
Neat! looks like gtest supports capturing the stdout to a string that we can return in the ret value across the FFI,
```
testing::internal::CaptureStdout();
// Run tests
// ...
std::string output = testing::internal::GetCapturedStdout();
```

Reply:
Unfortunately, it looks like stdout redirection is one of the disabled features in the Hexagon SDK gtest version.

Reply:
How so? in that case can we target a generic gtest and then modify to support for Hexagon as we have discussed potentially doing?

Reply:
My understanding is that one must set `GTEST_HAS_STREAM_REDIRECTION=0` when compiling gtest for Hexagon which [disables](https://github.com/google/googletest/blob/8ded48c37be09d8cc3665af1b414c5d53c0862e7/googletest/include/gtest/internal/gtest-port.h#L1130) the `CaptureStdout` function.  I believe this would be true even if we used a "stock" gtest version as opposed to the gtest version in the Hexagon SDK.

Reply:
Darn. @kparzysz-quic any idea why this is disabled for Hexagon? Is there an alternative for us to capture stdout on Hexagon?

Reply:
I don't know.  What happens if we set `GTEST_HAS_STREAM_REDIRECTION=1`?

Reply:
GTest stream redirection relies on dup2(), which does not exist under QuRT.

Reply:
This may be useful: https://google.github.io/googletest/advanced.html#extending-googletest-by-handling-test-events